src/nupic/algorithms/Connections.cpp

```diff
@@ -349,6 +352,82 @@ void Connections::computeActivity(
 }
 
 
+void Connections::adaptSegment(Segment segment, SDR &inputs,
```

Comment:
for clarity, make all but segment const

Reply:
Permanences are float32's so const doesn't do anything in this situation.  SDR's should never be made constant since their getter methods are not constant.  Their getter methods are non-constant because they can cause it to convert-format & cache the SDR's value.

Reply:
Read only proxies are pretty much free, but I don't want to force the user (the SP & TM class) to make a proxy here.  On the other hand, the user is free to pass in an SDR_Proxy if they want to, and the SP & TM & connections classes won't try to modify it.